### PR TITLE
Set config value for autoloader-suffix

### PR DESCRIPTION
**Pull request by TisVictress**
Is there a better string value for the suffix?

### CODE REVIEW

File: constants.go [hunk 1]
```diff
@@ -5,6 +5,9 @@ const (
 	ComposerGlobalLayerName   = "composer-global"
 	ComposerPhpIniLayerName   = "composer-php-ini"
 
+	// Autoloader Suffix
+	ComposerAutoloaderSuffix = "PaketoDefaultAutoloaderSuffix"
```
Comment (TisVictress) [inline constants.go:9]: Default suffix?

Comment (sophiewigmore) [reply constants.go:9]: ?

Comment (TisVictress) [reply constants.go:9]: @sophiewigmore Sorry for the confusion. I meant is this a good suffix value?

Comment (sophiewigmore) [reply constants.go:9]: ah. I think this is fine

**Comment by sophiewigmore** [comment]
@TisVictress this change seems good to me. Although you might need to rebase against main because the build code here looks a bit out of date (contains the `dumpautoloader` executable still). 

We also have [`composer global` behaviour](https://github.com/paketo-buildpacks/composer-install/blob/main/integration/global_test.go) during build, and I was wondering if those builds are reproducible also? Does it need access to the same configuration suffix?

**Comment by TisVictress** [comment]
> can you add a bit of context to the PR

@sophiewigmore No worries!

The `composer install` command generates `autoload*.php` files that contain a different checksum each time. So any image that gets rebuilt with this buildpack will be different even if there are no changes to the app, due to the random checksums.  

Here's a diff of two images built using the same app:
```
container-diff diff --type file daemon://compinstall1:latest daemon://compinstall2:latest

-----File-----

These entries have been added to compinstall1:latest: None

These entries have been deleted from compinstall1:latest: None

These entries have been changed between compinstall1:latest and compinstall2:latest:
FILE                                                                                                    SIZE1        SIZE2
/layers/paketo-buildpacks_composer-install/composer-packages/vendor/composer/autoload_static.php        8.7K         8.7K
/layers/paketo-buildpacks_composer-install/composer-packages/vendor/composer/autoload_real.php          1.7K         1.7K
/layers/paketo-buildpacks_composer-install/composer-packages/vendor/autoload.php                        771B         771B
```

This change sets the autoloader suffix to avoid the random checksum values mentioned above to ensure [that builds are reproducible](https://github.com/paketo-buildpacks/php/issues/568).

**Comment by TisVictress** [comment]
> I was wondering if those builds are reproducible also?

@sophiewigmore  They are reproducible. I've added another test.

**Review by sophiewigmore — APPROVED** [review]

**Comment by sophiewigmore** [comment]
@thitch97 I have held back on merging so you can take a look if you still want to

**Review by thitch97 — APPROVED** [review]

**Pull request merged**